ods-devenv/scripts/deploy.sh

```diff
@@ -0,0 +1,1563 @@
+#   atlassian_mysql_port
+#   atlassian_mysql_version
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function initialize_atlassian_bitbucketdb() {
+    local mysql_ip
+    mysql_ip=$(docker inspect -f '{{.NetworkSettings.IPAddress}}' ${atlassian_mysql_container_name})
+    echo "Setting up bitbucket database on ${mysql_ip}:${atlassian_mysql_port}."
+    echo "jiradbrpwd" | docker container run -i --rm mysql:${atlassian_mysql_version} mysql -h "${mysql_ip}" -u root -p -e \
+        "create database ${atlassian_bitbucket_db_name} character set utf8 collate utf8_bin; \
+        GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,REFERENCES,ALTER,INDEX,CREATE TEMPORARY TABLES on bitbucketdb.* TO 'bitbucket_user'@'%' IDENTIFIED BY 'bitbucket_password'; \
+        flush privileges;"
+}
+
+#######################################
+# The automated ODS setup requires some repositories to be existing in the
+# local bitbucket installation. This function takes care of it.
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function create_empty_ods_repositories() {
+    pushd ods-setup
+    ./bitbucket.sh --insecure \
+        --bitbucket "http://${public_hostname}:${atlassian_bitbucket_port}" \
+        --user openshift \
+        --password openshift \
+        --ods-project opendevstack
+    popd
+}
+
+#######################################
+# For each of the listed names this function will delete the corresponding
+# repository in the local BitBucket instance in the opendevstack project
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function delete_ods_repositories() {
+    for repository in ods-core ods-quickstarters ods-jenkins-shared-library ods-provisioning-app ods-configuration ods-document-generation-templates; do
+        echo "Deleting repository opendevstack/${repository} on http://${public_hostname}:${atlassian_bitbucket_port}."
+        curl -X DELETE --user openshift:openshift "http://${public_hostname}:${atlassian_bitbucket_port}/rest/api/1.0/projects/opendevstack/repos/${repository}"
+    done
+}
+
+#######################################
+# Makes use of ods-core/ods-setup-repos.sh to clone ods repositories from github
+# and push them to the local BitBucket instance.
+# Globals:
+#   atlassian_bitbucket_port
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function initialise_ods_repositories() {
+    local opendevstack_dir="/home/${USER}/opendevstack"
+
+    mkdir -p "${opendevstack_dir}"
+    pushd "${opendevstack_dir}"
+    curl -LO https://raw.githubusercontent.com/opendevstack/ods-core/${ods_git_ref}/ods-setup/repos.sh
+    chmod u+x ./repos.sh
+    ./repos.sh --init --confirm --source-git-ref "${ods_git_ref}" --target-git-ref "${ods_git_ref}" --bitbucket "http://openshift:openshift@${public_hostname}:${atlassian_bitbucket_port}" --verbose
```

Comment:
this should be the bitbucket cname - and internal port .... otherwise one has to change all the remotes if the IP changes ...

Reply:
yes, good point!